buildSrc/src/main/groovy/net/fabricmc/fabric/impl/gradle/ValidateModuleTask.groovy

```diff
@@ -0,0 +1,33 @@
+package net.fabricmc.fabric.impl.gradle
+
+import groovy.json.JsonSlurper
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.TaskAction
+
+class ValidateModuleTask extends DefaultTask {
+    @TaskAction
+    void validate() {
+        project.subprojects { subProject ->
```

Comment:
Crossing project boundaries seems to be discouraged by Gradle -- what you probably want to do is have `ValidateModuleTask` be registered on every subproject, and only act on that project. Then, through Gradle's workings running `./gradlew validateModule` would run the task in every subproject it's present in.

Reply:
No more interproject mingling now.